cmd/juju/application/deployer/charm.go

```diff
@@ -218,36 +218,13 @@ func (d *predeployedLocalCharm) PrepareAndDeploy(ctx *cmd.Context, deployAPI Dep
 		ctx.Infof("ignoring dry-run flag for local charms")
 	}
 
-	modelCfg, err := getModelConfig(deployAPI)
-	if err != nil {
-		return errors.Trace(err)
-	}
-
-	// Avoid deploying charm if it's not valid for the model.
-	base, err := corebase.GetBaseFromSeries(d.userCharmURL.Series)
-	if err != nil {
-		return errors.Trace(err)
-	}
-	if err := d.validateCharmBaseWithName(base, userCharmURL.Name, modelCfg.ImageStream()); err != nil {
-		return errors.Trace(err)
-	}
-
 	if err := d.validateCharmFlags(); err != nil {
 		return errors.Trace(err)
 	}
 
-	charmInfo, err := deployAPI.CharmInfo(d.userCharmURL.String())
-	if err != nil {
-		return errors.Trace(err)
-	}
 	ctx.Infof(formatLocatedText(d.userCharmURL, commoncharm.Origin{}))
-	checkPodspec(charmInfo.Charm(), ctx)
```

Comment:
Are we sure we want to remove this check?

Reply:
https://github.com/juju/juju/blob/3.6/cmd/juju/application/deployer/charm.go#L59-L69

We still do this check later